hive-metastore/src/test/java/org/apache/iceberg/hive/TestCachedClientPool.java

```diff
@@ -31,11 +38,106 @@ public void testClientPoolCleaner() throws InterruptedException {
     String metastoreUri = hiveConf.get(HiveConf.ConfVars.METASTOREURIS.varname, "");
     CachedClientPool clientPool = new CachedClientPool(hiveConf, Collections.emptyMap());
     HiveClientPool clientPool1 = clientPool.clientPool();
-    Assert.assertTrue(CachedClientPool.clientPoolCache().getIfPresent(metastoreUri) == clientPool1);
+    Supplier<Object> uri = () -> CachedClientPool.URIElement.of(metastoreUri);
+    Assert.assertTrue(
+        CachedClientPool.clientPoolCache()
+                .getIfPresent(CachedClientPool.toKey(Collections.singletonList(uri)))
+            == clientPool1);
     TimeUnit.MILLISECONDS.sleep(EVICTION_INTERVAL - TimeUnit.SECONDS.toMillis(2));
```

Comment:
this test methods takes **23+ seconds**, which is too long imo for a simple unit test. We might want to decrease the eviction interval for testing (can be a separate PR)